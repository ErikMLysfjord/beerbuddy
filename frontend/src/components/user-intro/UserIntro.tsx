import emoji from "/beerEmoji.svg";
import useWindowDimensions from "../../utils/useWindowDimensions";
import Logo from "../logo/Logo";
import styles from "./UserIntro.module.css";

/**
 * UserIntro component that greets a logged in user with their username.
 * @returns a UserIntro component
 */
const UserIntro = () => {
  const { width } = useWindowDimensions();
  if (width < 1000) {
    return (
      <>
        <header className={styles.headingWrapper}>
          <Logo />
        </header>
        <hr className={styles.separator} />
      </>
    );
  }
  return (
    <>
      <header className={styles.headingWrapper}>
        <h1 className={styles.heading}>
          <span className={styles.bold}>Welcome</span>,{" "}
          {localStorage.getItem("userNameBeerBuddy")
            ? localStorage.getItem("userNameBeerBuddy")
            : ""}
        </h1>
        <img className={styles.emoji} src={emoji} alt="Beer clinking Emoji" />
      </header>
      <hr className={styles.separator} />
    </>
  );
};
export default UserIntro;
