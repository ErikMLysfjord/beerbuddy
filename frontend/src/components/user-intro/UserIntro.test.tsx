import { describe, it, expect } from "vitest";
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import UserIntro from "./UserIntro";
import { act } from "react-dom/test-utils";
import { axe } from "jest-axe";

// Mock localStorage so it returns "Erik".
global.localStorage = {
  ...global.localStorage,
  getItem: () => "Erik",
};

describe("UserIntro", () => {
  it("is accessible", async () => {
    const { container } = render(<UserIntro />);
    expect(await axe(container)).toHaveNoViolations();
  });

  it("renders correctly", () => {
    const { container } = render(<UserIntro />);
    expect(container).toMatchSnapshot();
  });

  it("renders with correct heading", () => {
    const { getByRole } = render(<UserIntro />);
    expect(getByRole("heading")).toBeInTheDocument();
    expect(getByRole("heading")).toHaveTextContent("Welcome");

    expect(getByRole("heading")).toHaveTextContent("Erik");
  });

  it("renders with correct emoji", () => {
    const { getByAltText } = render(<UserIntro />);
    expect(getByAltText("Beer clinking Emoji")).toBeInTheDocument();
  });

  it("renders with correct separator", () => {
    const { getByRole } = render(<UserIntro />);
    expect(getByRole("separator")).toBeInTheDocument();
  });

  it("renders with no name", () => {
    // Mock localStorage so it returns an empty string.
    global.localStorage = {
      ...global.localStorage,
      getItem: () => "",
    };
    render(<UserIntro />);

    expect(screen.queryByText("Welcome")).toBeInTheDocument();
    expect(screen.queryByText("Erik")).not.toBeInTheDocument();
  });

  it("renders with correct text", () => {
    render(<UserIntro />);

    // Mock the window width to be 500px.
    // This is done to ensure the correct text is rendered.
    global.innerWidth = 500;
    act(() => {
      global.dispatchEvent(new Event("resize"));
    });

    expect(screen.queryByText("Welcome")).not.toBeInTheDocument();
  });
});
