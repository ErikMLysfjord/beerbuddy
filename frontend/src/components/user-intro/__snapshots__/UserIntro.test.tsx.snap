// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`UserIntro > renders correctly 1`] = `
<div>
  <header
    aria-label="User intro"
    class="_headingWrapper_95bd34"
  >
    <h1
      class="_heading_95bd34"
    >
      <b>
        Welcome
      </b>
      , 
      Erik
    </h1>
    <img
      alt="Beer clinking Emoji"
      class="_emoji_95bd34"
      src="/project2/beerEmoji.svg"
    />
  </header>
  <hr
    class="_separator_95bd34"
  />
</div>
`;
